fieldservice_sale/models/sale_order_line.py

```diff
@@ -54,12 +54,29 @@ def create(self, values):
             line.order_id._field_service_generation()
         return line
 
+    def _get_invoiceable_fsm_order_domain(self):
+        """
+        Override this method to define more search criteria for invoiceable
+        fsm order
+        :return:
+        """
+        invoiceable_stage_ids = self.env["fsm.stage"]._get_invoiceable_stage()
+        dom = [
+            ("sale_line_id", "=", self.id),
+            ("invoice_lines", "=", False),
```

Comment:
The test_sale_order_1 fails because this domain omits the FSM Order created by sale order so it doesn't get linked to the invoice that is created.  There is no SOL set on that FSM Order based on configuration.  I think this should fix it:

```suggestion
            "|', ("sale_line_id", "=", self.id), ("sale_id", "=", self.order_id.id),
            ("invoice_lines", "=", False),
```